server/products.js

```diff
@@ -0,0 +1,44 @@
+const db = require('APP/db')
+const Product = db.model('product')
+
+module.exports = require('express').Router()
+
+  .param('id', (req, res, next, id) => {
+    Product.findbyId(id)
+      .then(foundProduct => {
+        if (!foundProduct) throw Error('omg error 404')
+        req.product = foundProduct
+        next()
+      })
+      .catch(next)
+  })
+
+  .get('/:id', (req, res, next) => {
+    res.send(req.product)
+  })
+
+
+  .get('/', (req, res, next) => {
+    Product.findAll().then(allProducts => res.send(allProducts)).catch(next)
+  })
+
+  .post('/', (req, res, next) => {
+      Product.create(req.body)
+        .then(createdProduct => {
+          res.status(201).send(createdProduct)
+        })
+        .catch(next)
+  })
+
+  .put('/:id', (req, res, next) => {
+    req.product.update(req.body)
+      .then(updatedProduct => {
+        res.send(updatedProduct)
+      })
+      .catch(next)
+  })
+
+
+  .delete('/:id', (req, res, next) => {
+    req.product.destroy().then(destroyedItem => console.log('destroyed',destroyedItem)).catch(next)
```

Comment:
i am pretty sure this does not return anything, so it should just be a function that takes no args and does `res.status(204).send(`deletion successful`)`